index.js

```diff
@@ -1,230 +1,227 @@
 
-  if (timeDifference < 0) {
-    targetTime.setDate(targetTime.getDate() + 1);
-    timeDifference = targetTime - moment;
-  }
-  isOpen = moment.getHours() > 9 && moment.getHours() < 17;
+    if (timeDifference < 0) {
+      targetTime.setDate(targetTime.getDate() + 1);
+      timeDifference = targetTime - moment;
+    }
+    isOpen = moment.getHours() > 9 && moment.getHours() < 17;
 
-  setTimeout(() => {
-    alert(message);
+    setTimeout(() => {
+      alert(message);
 
-    openAndCloseAlert(hour, minute, second, message);
-  }, timeDifference);
-};
+      openAndCloseAlert(hour, minute, second, message);
+    }, timeDifference);
+  };
 
-openAndCloseAlert(9, 0, 0, "Market is open now!"); // Opening Time alert
-openAndCloseAlert(17, 0, 0, "Market is closed now!"); // Closing Time alert
+  openAndCloseAlert(9, 0, 0, "Market is open now!"); // Opening Time alert
+  openAndCloseAlert(17, 0, 0, "Market is closed now!"); // Closing Time alert
+
+  const pushAlertToList = document.getElementById("alert-currency-form");
+  pushAlertToList.addEventListener("submit", async (event) => {
+    event.preventDefault();
+    const alertCurrency = document.getElementById("to-currency-alert").value;
+    let alertRate = document.getElementById("alert-rate").value;
+    const objAlert = {
+      code: alertCurrency,
+      rate: alertRate,
+    };
+    alertCurrencies.push(objAlert);
+    console.log(alertCurrencies);
+  });
 
-const pushToAlertList = (event) => {
-  event.preventDefault();
-  const alertCurrency = document.getElementById("to-currency-alert").value;
-  let alertRate = document.getElementById("alert-rate").value;
-  const objAlert = {
-    code: alertCurrency,
-    rate: alertRate,
-  };
-  alertCurrencies.push(objAlert);
-  console.log(alertCurrencies);
-};
-
-function observer() {
-  for (let i = 0; i < alertCurrencies.length; i++) {
-    const alertCurrency = alertCurrencies[i];
-    const currentRate = currencies.rates.find((rate) => rate.code === alertCurrency.code).rate;
-
-    if (currentRate !== undefined) {
-      if (currentRate == alertCurrency.rate) {
-        alert(
-          `Alert: ${alertCurrency.code} has reached the alert rate of ${alertCurrency.rate}. Current rate: ${currentRate}`
-        );
-        alertCurrencies.splice(i, 1); // Remove the alert from the list
-        i--; // Adjust the index after removal
+  function observer() {
+    for (let i = 0; i < alertCurrencies.length; i++) {
+      const alertCurrency = alertCurrencies[i];
+      const currentRate = myData.rates.find((rate) => rate.code === alertCurrency.code).rate;
+
+      if (currentRate !== undefined) {
+        if (currentRate == alertCurrency.rate) {
+          alert(
+            `Alert: ${alertCurrency.code} has reached the alert rate of ${alertCurrency.rate}. Current rate: ${currentRate}`
+          );
+          alertCurrencies.splice(i, 1); // Remove the alert from the list
+          i--; // Adjust the index after removal
```

Comment:
While this works and is logically true, I would suggest constructing a new array (using `filter()`) without the removed values. In general, it's not a good idea to modify an array _while_ it is being iterated.